### PR TITLE
Migrate common/test_api_utils.py from API 1.1 to 2.0

### PR DESCRIPTION
**Pull request by gpaulos**
This is an interim fix for the library file common/test_api_utils.py which was entirely API 1.1

This is a legacy library that is used infrequently but is called in the Smoke test tests/compute/test_rackhd20_computenode_pollers.py, so it needs immediate revision.

Here is the rundown of changes:
- replaced all API 1.1 to 2.0 and made necessary logic changes to support 2.0 JSON
- removed and redirected get_active_workflow() and delete_active_workflow() which was duplicated in fit_common check_active workflow() and cancel_active_workflow()
- Fixed Hound violations

Still to do:
- Change VERBOSITY logging to FIT logging
- Better error recovery

@hohene @stuart-stanley @jimturnquist @johren

### CODE REVIEW

**Comment by gpaulos** [comment]
test this please

File: test/common/test_api_utils.py [hunk 1]
```diff
@@ -1,61 +1,60 @@
...
     sku = ""
     # get node info
-    mondata = fit_common.rackhdapi("/api/1.1/nodes/"+nodeid)
+    mondata = fit_common.rackhdapi("/api/2.0/nodes/" + nodeid)
     if mondata['status'] == 200:
         # get the sku id contained in the node
         sku = mondata['json'].get("sku")
         if sku:
-            skudata = fit_common.rackhdapi("/api/1.1/skus/"+sku)
+            skudata = fit_common.rackhdapi(sku)
```
Comment (johren) [inline test/common/test_api_utils.py]: Shouldn't this be /api/2.0/skus/sku?

File: test/common/test_api_utils.py [hunk 2]
```diff
@@ -341,29 +304,30 @@ def run_ipmi_command(node_ip, str_command, dict_credential):
     :return: result of remote_shell
     """
     remote_ssh_res = {'stdout': "", 'exitcode': 0}
-    str_ipmi_cmd = "ipmitool -I lanplus -H " \
-                   + node_ip + " -U " \
-                   + dict_credential['user'] \
-                   + " -P " + dict_credential['password'] \
-                   + " " + str_command
+    str_ipmi_cmd = "ipmitool -I lanplus -H " +\
```
Comment (johren) [inline test/common/test_api_utils.py]: nit:  I don't think the backslashes are needed at the end of these lines

File: test/common/test_api_utils.py [hunk 3]
```diff
@@ -379,52 +343,55 @@ def run_ipmi_command_to_node(node_id, str_ipmi_cmd):
...
-                poller_dict[poller_name] = {'poller_id':poller_id, 'poller_interval':poller_interval}
+                poller_dict[poller_name] = {'poller_id': poller_id, 'poller_interval': poller_interval}
 
     return poller_dict
 
+
 def get_ora_poller_id_list():
-    '''
+    """
     Get all poller ids for an onrack system
```
Comment (johren) [inline test/common/test_api_utils.py]: nit:  onrack?

**Comment by johren** [comment]
Couldn't we just slip tests/redfish10/test_redfish10_system_reset_base.py into this PR to remove the duplicate "get_rackhd_nodetype" function?

File: test/common/test_api_utils.py (hunk 1 above)
Comment (hohene) [inline test/common/test_api_utils.py]: The line 56 sets the sku to the full link which includes the api/2.0 in it.
"sku": "/api/2.0/skus/621828d7-1b7e-4d06-a224-326f4680ac9e

**Comment by hohene** [comment]
@johren In the PR for removing 1.1 FIT, I'll add the updates for get_rackhd_nodetype into tests/redfish10/test_redfish10_system_reset_base.py.  I'm fixing a couple more scripts that have the same thing.

File: test/common/test_api_utils.py (hunk 1 above)
Comment (johren) [inline test/common/test_api_utils.py]: Ah, I see.  This behavior changed.

**Comment by johren** [comment]
test this please

**Comment by keedya** [comment]
test this please

**Pull request merged**